Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryDeferredRequestAuditTransformsTest.java

```diff
@@ -0,0 +1,222 @@
+
+                    return webContextProperties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS);
+                }
+                return AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS;
+            }
+
+            @Override
+            protected String getWebServiceUrlFromRemoteObject(NhinTargetSystemType target, String serviceName) {
+                return remoteObjectUrl;
+            }
+        };
+
+        AssertionType assertion = createAssertion();
+        LogEventRequestType auditRequest = transforms.transformRequestToAuditMsg(TestPatientDiscoveryMessageHelper.
+            createPRPAIN201305UV02Request("Gallow", "Younger", "M", "01-12-2967", "1.1", "D123401", "2.2",
+                "abd3453dcd24wkkks545"), assertion, createNhinTarget(), NhincConstants.AUDIT_LOG_INBOUND_DIRECTION,
+            NhincConstants.AUDIT_LOG_ENTITY_INTERFACE, Boolean.TRUE, webContextProperties,
+            NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME);
+
+        testGetEventIdentificationType(auditRequest, NhincConstants.PATIENT_DISCOVERY_DEFERRED_REQ_SERVICE_NAME,
+            Boolean.TRUE);
+        testGetActiveParticipantSource(auditRequest, Boolean.TRUE, webContextProperties, localIp);
+        testGetActiveParticipantDestination(auditRequest, Boolean.TRUE, webContextProperties, remoteObjectUrl);
+        testCreateActiveParticipantFromUser(auditRequest, Boolean.TRUE, assertion);
+        assertNotNull("ParticipantObjectIdentification is null", auditRequest.getAuditMessage().
+            getParticipantObjectIdentification());
+        assertFalse("ParticipantObjectIdentification is empty", auditRequest.getAuditMessage().
+            getParticipantObjectIdentification().isEmpty());
+        assertEquals("ParticipantObjectIdentification is empty", auditRequest.getAuditMessage().
+            getParticipantObjectIdentification().size(), 2);
+        assertPatientParticipantObjectIdentification(auditRequest.getAuditMessage().getParticipantObjectIdentification()
+            .get(0));
+        assertQueryParticipantObjectIdentification(auditRequest.getAuditMessage().getParticipantObjectIdentification().
+            get(1));
+    }
+
+    @Test
+    public void transformResponseToAuditMsg() throws ConnectionManagerException, UnknownHostException {
+        final String localIp = "10.10.10.10";
+        final String remoteIp = "16.14.13.12";
+        final String remoteObjectUrl = "http://" + remoteIp + ":9090/source/AuditService";
+        final String wsRequestUrl = "http://" + remoteIp + ":9090/AuditService";
+
+        Properties webContextProperties = new Properties();
+        webContextProperties.setProperty(NhincConstants.WEB_SERVICE_REQUEST_URL, wsRequestUrl);
+        webContextProperties.setProperty(NhincConstants.REMOTE_HOST_ADDRESS, remoteIp);
+        webContextProperties.setProperty(NhincConstants.INBOUND_REPLY_TO,
+            "http://www.w3.org/2005/08/addressing/anonymous");
```

Comment:
Can we use NhincConstants.WSA_REPLY_TO as there is already a constant defined for this URL?